### PR TITLE
Refactor(eos_cli_config_gen): Wildcard dict to list for peer_filters

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

`peer_filters`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update all `host_vars` under molecule scenario `eos_cli_config_gen_v4.0` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Run molecule `cd ansible_collections/arista/avd/molecule ; make cli-4.0-schema`
  - [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1 (Claus):
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Tony
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.ini [hunk 1]
```diff
@@ -67,6 +67,7 @@ mpls
 none_configuration
 ntp
 patch-panel
+peer_filters
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.ini]: we use `-` in hostnames for molecule. Please update files (clean up the ones with `_`)
this new "host" should also be added in `hosts.ini` in 4.0 molecule.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/peer_filters.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+### peer filters ###
+
+peer_filters:
+  PF1:
+    sequence_numbers:
+      10:
+        match: "as-path"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/peer_filters.yml]: change to `match: "as-range 1-2 result reject"`

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/peer_filters.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+      20:
+        match: "as-range"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/peer_filters.yml]: change to `match: "as-range 1-100 result accept"`

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/peer_filters.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  PF2:
+    sequence_numbers:
+      30:
+        match: "community-list"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/peer_filters.yml]: change to `match: "as-range 65000 result accept"`

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/inventory/host_vars/peer_filters.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+### peer filters ###
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/inventory/host_vars/peer_filters.yml]: update similar to the other peer-filters.yml file

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**